src/main/kotlin/org/rust/ide/intentions/DemorgansLawIntention.kt

```diff
@@ -0,0 +1,107 @@
+package org.rust.ide.intentions
+
+import com.intellij.codeInsight.intention.PsiElementBaseIntentionAction
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.tree.IElementType
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.util.elementType
+import org.rust.lang.core.psi.util.parentOfType
+
+class DemorgansLawIntention : PsiElementBaseIntentionAction() {
+    override fun getText() = "DeMorgan Law"
```

Comment:
Currently, yes. But I think this pattern could be useful in the future. You can merge that class if you like, it's no problem for me :smile: